### PR TITLE
CDAP-14442 remove avro and parquet dependencies

### PR DESCRIPTION
**Pull request by albertshau**
Formats are now in plugins. Removing them from the s3 plugins to
avoid classloading issues and reduce the jar sizes.

### CODE REVIEW

**Comment by albertshau** [comment]
On Hold -- issues came up during testing

File: pom.xml [hunk 1]
```diff
@@ -77,6 +77,16 @@
       <groupId>co.cask.cdap</groupId>
       <artifactId>cdap-formats</artifactId>
       <version>${cdap.version}</version>
+      <exclusions>
+        <exclusion>
+          <groupId>org.apache.avro</groupId>
+          <artifactId>avro</artifactId>
+        </exclusion>
+        <exclusion>
+          <groupId>io.thekraken</groupId>
```
Comment (CuriousVini) [inline pom.xml]: why do we need this change?

File: pom.xml [hunk 2]
```diff
@@ -196,33 +206,6 @@
...
-      </exclusions>
-    </dependency>
-    <dependency>
-      <groupId>org.apache.avro</groupId>
-      <artifactId>avro-mapred</artifactId>
-      <classifier>hadoop2</classifier>
-      <version>1.7.7</version>
-    </dependency>
-    <dependency>
-      <groupId>xerces</groupId>
```
Comment (CuriousVini) [inline pom.xml]: Is this change intentional?

File: pom.xml (hunk 1 above)
Comment (albertshau) [reply pom.xml]: we don't need to exclude it, but it reduces the size of the jar. This dependency will be removed from CDAP when stream formats are removed.

File: pom.xml (hunk 2 above)
Comment (albertshau) [reply pom.xml]: yeah, it's moved into the format plugins in https://github.com/caskdata/hydrator-plugins/pull/860. This shouldn't be in the plugin at all, but is required to workaround https://issues.cask.co/browse/CDAP-14562
